src/fountain.rs

```diff
@@ -267,95 +348,11 @@ impl std::fmt::Display for Part {
 }
 
 impl Part {
-    pub fn from_cbor(cbor: Vec<u8>) -> anyhow::Result<Self> {
-        let mut decoder = cbor::Decoder::from_bytes(cbor);
-        let items: Vec<cbor::Cbor> = match decoder.items().collect::<Result<_, _>>() {
-            Ok(i) => i,
-            Err(_) => return Err(anyhow::anyhow!("invalid cbor serialization for Part")),
-        };
-        if items.len() != 1 {
-            return Err(anyhow::anyhow!("invalid cbor length for Part"));
+    pub fn from_cbor(cbor: &[u8]) -> anyhow::Result<Self> {
+        match serde_cbor::from_slice(cbor) {
+            Ok(v) => anyhow::Result::Ok(v),
```

Comment:
The `anyhow::Result::` prefixes should not be necessary in this and the following line.